src/commands/environmentVariables/editEnvironmentVariables/EnvironmentVariablesEditDraftStep.ts

```diff
@@ -21,10 +22,17 @@ export class EnvironmentVariablesEditDraftStep<T extends EnvironmentVariablesEdi
     public async execute(context: T, progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<void> {
         progress.report({ message: localize('editingEnv', 'Editing environment variables (draft)...') });
         this.revisionDraftTemplate.containers ??= [];
-
         const container: Container = this.revisionDraftTemplate.containers[context.containersIdx] ?? {};
-        container.env = context.environmentVariables;
 
+        const envMap: Map<string, EnvironmentVar> = new Map();
+        for (const env of container.env ?? []) {
```

Comment:
Dunno if it impacts perf or not, but you could also just make a new Map from the JSON object, and then overwrite with the for of statement below.

I think it's more readable since it's a little confusing at first seeing both of these loops, but that may just be me.